### PR TITLE
fix(ui): preserve administration plugins and restore chart navigation

### PR DESCRIPTION
**Pull request by yingbull**
Administration fragments could break later screens: the OHIP simulation page attached a GET iframe handler to every persistent navigation link, replacing Age-Sex Report’s required POST. eForm fragments reloaded jQuery and discarded validation plugins because the real AJAX header contains two XMLHttpRequest markers. Remove the stray billing handler, recognize comma-separated AJAX markers, and limit the eForm footer handler to its own dynamic content. Keep Patient List variables local and initialize Visit Report’s actual form ID.

Restore the calculator entry in the current chart and correct four calculator stylesheet paths. The new header link passes the existing record reference so the authenticated demographic manager resolves prefilled age/sex server-side; neither value is included in the header popup or fallback URL. Supply Row Display’s CSRF bootstrap/readiness and explicit error handling; correct the lot-search form handler.

Validation: the completed installed-package administration sweep visited 101 items (two explicit skips), isolating nine observations to Age-Sex, Visit and Overnight. A regression running the actual OHIP initializer fails on the original global handler and passes after removal. Real UI/header probes prove the duplicate-marker cause. The latest fixes await the next package install after host LXD recovery. Chart navigation (11 opened/13 policy skips) and both real lot searches pass. Calculator numerical scenarios completed but strict recording exposed the now-corrected stylesheet URLs; final clean retest is pending.

Review follow-ups `ee3d068b71` and `6c75d801a2` add complete JSP contracts and a calculator-link regression that fails against the old header and passes after the fix. All 679 combined Node cases pass. [Current-head CI](https://github.com/carlos-emr/carlos/actions/runs/35209972118) is green: 12,139 Java tests, zero failures/errors, 48 skips, and successful JSP compilation. CodeRabbit completed its [full review](https://github.com/carlos-emr/carlos/pull/3691#pullrequestreview-5234064316), accepted the [fixes](https://github.com/carlos-emr/carlos/pull/3691#issuecomment-5712724148) and [final documentation clarification](https://github.com/carlos-emr/carlos/pull/3691#issuecomment-5712774883); all review threads are resolved. Existing legacy child-calculator URL contracts are unchanged.

Addresses #3682. Based on and targets `release/2026.08`; no schema/Flyway change. Shared workflow/validation ledger: #3693.

## Summary by Sourcery

Preserve administration workflows and restore reliable chart calculator navigation without exposing patient attributes in URLs.

New Features:
- Restore calculator access from the current chart with server-side demographic resolution for the originating patient.

Bug Fixes:
- Prevent administration fragments from hijacking persistent navigation or replacing shell jQuery and validation plugins.
- Correct report form initialization, patient-list scoping, lot-number search handling, calculator stylesheet paths, and lab row-display request readiness and error handling.

Enhancements:
- Improve dynamic administration and report fragment behavior while preserving existing navigation, plugin, CSRF, and legacy calculator URL contracts.

Tests:
- Add regressions covering OHIP navigation isolation and calculator-link demographic URL handling.

### CODE REVIEW

**Comment by yingbull** [comment]
@coderabbitai review

**Comment by yingbull** [comment]
@coderabbitai review

File: scripts/admin-fragment-navigation.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const assert = require('node:assert/strict');
+const test = require('node:test');
+const fs = require('node:fs');
+const path = require('node:path');
+const vm = require('node:vm');
+
+test('loading OHIP simulation initializes its own form without hijacking shell navigation', () => {
+  const source = fs.readFileSync(path.join(__dirname,
+    '../src/main/webapp/WEB-INF/jsp/billing/CA/ON/billingOHIPsimulation.jsp'), 'utf8');
+  const scripts = [...source.matchAll(/<script\b[^>]*>([\s\S]*?)<\/script>/gi)];
```
[the remark this answers is not in the document]
Comment (yingbull) [reply scripts/admin-fragment-navigation.test.js]: Addressed in 7202c1e791. The regression now extracts the known initializer using its exact source markers, with explicit bounds assertions, rather than using an HTML-matching regular expression. It still executes the real initializer and rejects added shell navigation handlers; the test passes.

**Comment by yingbull** [comment]
@coderabbitai review

Please review the latest fixes and regressions. The PR description records completed validation and the final VM checks still pending host LXD recovery.

File: scripts/admin-fragment-navigation.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const path = require('node:path');
+const vm = require('node:vm');
+
+test('loading OHIP simulation initializes its own form without hijacking shell navigation', () => {
+  const source = fs.readFileSync(path.join(__dirname,
+    '../src/main/webapp/WEB-INF/jsp/billing/CA/ON/billingOHIPsimulation.jsp'), 'utf8');
+  const marker = "registerFormSubmit('serviceform'";
+  const markerAt = source.indexOf(marker);
+  const openingTag = '<script type="text/javascript">';
+  const start = source.lastIndexOf(openingTag, markerAt);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply scripts/admin-fragment-navigation.test.js:14]: Reviewed as a false positive in a Node-only regression test. `markerAt`, `start` and `end` are numeric offsets computed from a fixed repository JSP, used to extract its known initializer for vm.runInNewContext. They are not request input and are never emitted into browser HTML. No XSS sink or sanitizer exists on this path; changing application encoding or weakening the regression is not warranted.

File: scripts/admin-fragment-navigation.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  const end = source.indexOf('</script>', markerAt);
+  assert.ok(markerAt >= 0 && start >= 0 && end > markerAt,
+    'The billing simulation initialization script was not found');
+  const script = source.slice(start + openingTag.length, end);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply scripts/admin-fragment-navigation.test.js:18]: Reviewed as a false positive in a Node-only regression test. `markerAt`, `start` and `end` are numeric offsets computed from a fixed repository JSP, used to extract its known initializer for vm.runInNewContext. They are not request input and are never emitted into browser HTML. No XSS sink or sanitizer exists on this path; changing application encoding or weakening the regression is not warranted.

[the remark this answers is not in the document]
Comment (yingbull) [reply scripts/admin-fragment-navigation.test.js:18]: Reviewed as a false positive in a Node-only regression test. `markerAt`, `start` and `end` are numeric offsets computed from a fixed repository JSP, used to extract its known initializer for vm.runInNewContext. They are not request input and are never emitted into browser HTML. No XSS sink or sanitizer exists on this path; changing application encoding or weakening the regression is not warranted.

**Comment by yingbull** [comment]
@coderabbitai full review

**Comment by yingbull** [comment]
@coderabbitai full review

**Comment by yingbull** [comment]
@coderabbitai Please analyze commit ee3d068b71 against full review 5234064316, including all three outside-diff documentation findings and inline comment 4035535508.

The header's fallback href and ClinicalCalculators popup now pass only the existing `demo` record reference, not age or sex. This uses the same route contract already used by `encounter/oscarMeasurements/Measurements.jsp`. In `encounter/calculators.jsp`, that reference resolves age/sex through `DemographicData.getDemographic(LoggedInInfo, demo)` and `DemographicManager.getDemographic`, which performs the patient-specific READ privilege check before loading the record. It preserves prefill for the originating chart without a fallback to whichever patient a shared EctSessionBean later happens to contain. The existing calculator menu and legacy child-calculator input contract are unchanged; this commit removes clinical values from the newly added header URL.

The new executable navigation regression renders synthetic values into both attributes, executes the actual onclick handler, and verifies the popup/fallback use the originating record reference and contain no age/sex parameters. It fails against the old header and passes after the fix; the existing OHIP fragment regression still passes. Added/expanded purpose, features, parameters and @since blocks for all twelve touched JSPs, preserving the existing valid comments. Please verify the endpoint contract and documentation accuracy, and report any remaining valid finding. Current-head Java/JSP CI is running; the review thread remains open until verification and green CI.

**Comment by yingbull** [comment]
@coderabbitai Please verify the documentation-only follow-up `6c75d801a2` against the remaining wording finding in your analysis https://github.com/carlos-emr/carlos/pull/3691#issuecomment-5712724148.

The CumulativeLabValues.jsp block now distinguishes CSRF readiness/token alerts from rendering successful and failed response bodies in the new row container. The latter behavior comes from `share/javascript/carlos-ajax.js`: `updater()` sets both successContainer and failureContainer to a DOM-element argument (lines 376–399), and installs its own onFailure wrapper to insert transport.responseText into failureContainer (lines 425–430). Thus the call does have shared failure rendering even without its own callback. This commit clarifies the documentation and changes no runtime code. The earlier calculator/other JSP fixes were already accepted in your analysis; 679 combined Node cases pass for their code, and current Java/JSP CI will verify the final commit. Please report whether any actionable finding remains.

File: src/main/webapp/WEB-INF/jsp/casemgmt/newEncounterHeader.jsp [hunk 4]
```diff
@@ -113,6 +113,10 @@ function fallbackCopy(text) {
 </div>
 
 <div id="header-bottom-row">
+    <div>
+        <a href="${carlos:forHtmlAttribute(ctx)}/encounter/ViewCalculators?sex=${carlos:forUriComponent(popupPatientSex)}&amp;age=${carlos:forUriComponent(popupPatientAge)}"
+           onclick="window.open('${carlos:forJavaScriptAttribute(ctx)}/encounter/ViewCalculators?sex=${carlos:forUriComponent(popupPatientSex)}&amp;age=${carlos:forUriComponent(popupPatientAge)}', 'ClinicalCalculators', 'width=800,height=650,scrollbars=yes,resizable=yes'); return false;"><fmt:message key="encounter.Index.calculators"/></a>
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/casemgmt/newEncounterHeader.jsp]: Fixed in ee3d068b71: both header links now pass only the originating chart reference to the existing permission-checked calculator route, which loads age and sex server-side. The executable regression fails against the prior header and passes with the fix; all 679 combined Node cases pass. CodeRabbit accepted the fix in #issuecomment-5712724148 and the final documentation clarification in #issuecomment-5712774883. Current head 6c75d801a2 has all CI green, including 12,139 Java tests (0 failures/errors, 48 skips) and JSP compilation: https://github.com/carlos-emr/carlos/actions/runs/35209972118 . Resolving this verified finding. Existing legacy child-calculator URL contracts are unchanged.

**Comment by yingbull** [comment]
@phc007 Ping for awareness.
